sdk-src/src/main/java/com/hpe/adm/nga/sdk/network/google/GoogleHttpClient.java

```diff
@@ -216,15 +224,26 @@ private OctaneHttpResponse execute(OctaneHttpRequest octaneHttpRequest, int retr
 
         try {
             httpResponse = executeRequest(httpRequest);
-            return convertHttpResponseToOctaneHttpResponse(httpResponse);
+            final OctaneHttpResponse octaneHttpResponse = convertHttpResponseToOctaneHttpResponse(httpResponse);
+            final String eTag = httpResponse.getHeaders().getETag();
+            if (eTag != null) {
+                requestToEtagMap.put(octaneHttpRequest, eTag);
+                cachedRequestToResponse.put(octaneHttpRequest, octaneHttpResponse);
+            }
+            return octaneHttpResponse;
         } catch (HttpResponseException e) {
 
+            final int statusCode = e.getStatusCode();
+            if (statusCode == HttpStatusCodes.STATUS_CODE_NOT_MODIFIED) {
```

Comment:
Ohh, so this is how it works, I thought it the cache was server side, and it just helps the server fetch the data faster...